include/signalrclient/signalr_client_config.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#pragma once
+
+#include <memory>
+#include "_exports.h"
+
+namespace web
+{
+    class web_proxy;
+    class credentials;
+    namespace http
+    {
+        namespace client
+        {
+            class http_client_config;
+        }
+    }
+    namespace websockets
+    {
+        namespace client
+        {
+            class websocket_client_config;
+        }
+    }
+}
+
+
+namespace signalr
+{
+    //If you need access to the underlying configuration objects, you need to include
+    //cpprest/http_client.h and/or cpprest/ws_client.h
+    class signalr_client_config
```

Comment:
Yes, that would surely make for a nicer API. Would you keep `(hub_)connection::set_headers(const std::unordered_map<utility::string_t, utility::string_t>& headers)`, or do you think it can be removed?